[5.0] Replace hard coded img element by JLayout joomla.html.image and DRY

hans2103 opened the pull request:
Pull Request for no Issue

### Summary of Changes

This PR will change the hard coded `<img src=""...` inside the default view of mod_banners by a call to JLayout joomla.html.image and keeps it DRY.

### Testing Instructions

- Joomla 5 with testing sample data. (adjust mod_banners config and unset the chosen client, otherwise you don't see images... https://github.com/joomla/joomla-cms/issues/40616)
- Go to frontend index.php/banners 

### Actual result BEFORE applying this Pull Request

You will see a banner image on the page. 

### Expected result AFTER applying this Pull Request

You will see a banner image on the page.  The same as BEFORE. The HTML output is also the same as BEFORE. 
The only difference is that it is rendered by JLayout which allows users to create an override. 

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

hans2103 committed:
Replace hard coded img element by JLayout joomla.html.image and DRY

dgrammatiko commented on modules/mod_banners/tmpl/default.php:
```diff
@@ -39,6 +40,17 @@
                 <?php $alt = $alt ?: Text::_('MOD_BANNERS_BANNER'); ?>
+                <?php $imageAttributes = [
+                    'src' => $baseurl . $imageurl,
+                    'alt' => htmlspecialchars($alt, ENT_QUOTES, 'UTF-8')
```
The `alt` is escaped in the Layout so you could pass it directly here

hans2103 replied:
adjusted my PR

hans2103 committed:
alt is escaped in JLayout

brianteeman commented on modules/mod_banners/tmpl/default.php:
```diff
@@ -39,6 +40,17 @@
                 <?php $alt = $item->params->get('alt'); ?>
                 <?php $alt = $alt ?: $item->name; ?>
                 <?php $alt = $alt ?: Text::_('MOD_BANNERS_BANNER'); ?>
```
Does this line ever get reached. Surely the previous condition is always matched

hans2103 replied:
I was wondering the same thing. I think that is out of scope of this PR and adjusting / removing the line should be in another PR. Agree?

brianteeman replied:
yes I realised that after typing it ;)

hans2103 replied:
should be something like below I guess

```
$alt = $item->params->get('alt', $item->name); 
```

hans2103 replied:
but... that's up to another PR. ;-)

wojtekLs commented on modules/mod_banners/tmpl/default.php:
```diff
@@ -39,6 +40,17 @@
                 <?php $alt = $item->params->get('alt'); ?>
                 <?php $alt = $alt ?: $item->name; ?>
                 <?php $alt = $alt ?: Text::_('MOD_BANNERS_BANNER'); ?>
+                <?php $imageAttributes = [
```
why there are multiple `php` block there?

richard67 replied:
Because we do it like that in our layouts so we have an HTML code style with small PHP blocks.

wojtekLs commented on modules/mod_banners/tmpl/default.php:
```diff
@@ -47,16 +59,7 @@
-                                    echo 'height="' . $height . '"';
-                                } ?>
-                            >
+                            <?php echo $image; ?>
```
LGTM :)

wojtekLs reviewed (changes requested):
Please merge multiple `php` blocks into one. Same goes with `if` statements.

richard67 commented:
> Please merge multiple `php` blocks into one. Same goes with `if` statements.

Maybe you should first study our code style before making recommendations as if you were a long term contributor.

hans2103 committed:
Merge branch '5.0-dev' into feature/5.0-dev/replace-img-by-jlayout-for-mod-banners

hans2103 committed:
reduce code

hans2103 committed:
fix reference

chmst commented:
Looks good to me. The $alt seems to be not correct. but this is not in scope of this PR.

richard67 commented:
> > code style
> 
> @richard67 No man. Having formalized code style doesnt mean that its good code style. Nor does it mean that it should be blindly followed.

@wojtekLs Better than following your comments which are completely useless since you talk about stuff of which you have no knowledge, e.g. our rules.

richard67 commented:
https://developer.joomla.org/coding-standards/php-code.html

> Mixed language usage (e.g. at the layout files)
>
> For layout files and all files where we use a mix of PHP and HTML (all PHP files in the view/tmpl and layout folder) we additionally wrap every line into a <?php ... ?> block and use the alternative syntax for control structures. This should make the code easier to read and make it easier to move blocks around without creating fatal errors due to missing <?php ... ?> tags.

@wojtekLs I haven't made these rules. For changing these rules it needs a decision by the production department. We all know that, only you insist on violating these rules because you comment here without having made yourself familiar with our rules.

brianteeman commented:
Why do we have to put to with this troll

hans2103 commented:
Please, can we get the focus on the PR again? The discussion about code styling is out of scope for this issue.
Thank you

Quy commented on modules/mod_banners/tmpl/default.php:
```diff
@@ -31,14 +32,23 @@
+                    'src' => $baseurl . $imageurl,
+                    'alt' => $alt
+                ]; ?>
+                <?php if (!empty($item->params->get('width'))) {
+                    $imageAttributes['width'] = $item->params->get('width');
+                } ?>
+                <?php if (!empty($item->params->get('height'))) {
+                    $imageAttributes['height'] = $item->params->get('height');
+                } ?>
```
```suggestion
                <?php if (!empty($item->params->get('width'))) : ?>
                    <?php $imageAttributes['width'] = $item->params->get('width'); ?>
                <?php endif; ?>
                <?php if (!empty($item->params->get('height'))) : ?>
                    <?php $imageAttributes['height'] = $item->params->get('height'); ?>
                <?php endif; ?>
```

hans2103 committed:
Update modules/mod_banners/tmpl/default.php

Co-authored-by: Quy <quy@nomonkeybiz.com>

QuyTon committed:
Merge branch '5.0-dev' into feature/5.0-dev/replace-img-by-jlayout-for-mod-banners

Quy commented:
The width/height settings are ignored.

Before PR:
```
<img
                                src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg#joomlaImage://local-images/banners/shop-ad-books.jpg?width=468&height=60"
                                alt="Banner"
                                width="50"                                height="50"                            >
```

After PR:
```
<img src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg" alt="Banner" width="468" height="60" loading="lazy">
```

HLeithner commented:
This pull request has been automatically rebased to 5.1-dev.

wojtekLs commented:
@HLeithner reasons?

HLeithner commented:
all pull requests have been rebased because the new branch for Joomla 5.1 has been created. since Joomla 5.0 don't get any new features from this point we we automatically rebase Pull requests with the label feature to the next minor branch. and bugs to the olders supported branch (in this case 4.4)

MacJoom committed:
Joomla! 4.4.0 Rc 1

MacJoom committed:
Revert to dev

Signed-off-by: Martin Kopp <kopp05@infotech.ch>

MacJoom committed:
Update com_joomlaupdate.ini (#42047)

Remove sentence from Potential Upgrade Issue.

brianteeman committed:
[5.0] Sample Data Titles contrast a11y (#42054)

dgrammatiko committed:
[5.0] Console error due to missing dependency (#42058)


Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

brianteeman committed:
[5.0] Installer drag'n drop dark mode a11y (#42053)

brianteeman committed:
[5.0] Treeselect dark mode (#42057)

* [5.0] Treeselect dark mode

* tree line

brianteeman committed:
[5.0] Dark Mode: Versions toolbar (#42051)


Pull request for #42013

* change to use btn-primary

---------

Co-authored-by: Quy <quy@nomonkeybiz.com>

brianteeman committed:
[5.0] Remove bs2 classes (#42066)

This button doesnt appear to be displayed but it is presumably there for a reason

The classes on the button appear to be a remnant from nik's own j3 implementation of this as two of the three are bs2 classes that are not present in bs5 and can be safely removed as they do nothing

Thanks to @wilsonge for pointing this out

Testing only by code review

brianteeman committed:
[5.0] testing data rss fix (#42064)

The testing sample data is not distributed with Joomla so it can be merged in to the main branch

dgrammatiko committed:
[5.0] Fix the calendar (#42060)

* again

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

* missing fallback

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

---------

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

brianteeman committed:
[5.0] com_templates buttons dark mode (#42071)

Adds a btn-primary class to the create child template and copy template buttons

brianteeman committed:
[5.0] form-select background (#42078)

remove the white background for form-select[multiple]

brianteeman committed:
[5.0] code-color dark mode (#42073)

* [5.0] code-color dark mode

Overrides the $pink variable from bootstrap to improve contrast and ensure it meets accessibility contrast standards

* second try

Fedik committed:
[5.0] Fix dialog height in Safari (#42079)

* Fix dialog height in Safari

* Fix dialog height in Safari

richard67 committed:
Fix SCSS CS broken by PR #42078 (#42089)

QuyTon committed:
Change button style (#42085)

Hackwar committed:
Updating Joomla framework to version 3.0.0 (#42091)

Fedik committed:
Check if the value consist with id:alias, extract the id only (#42096)

Fedik committed:
Fix language association "clear" button (#42098)

bembelimen committed:
Merge remote-tracking branch 'origin/4.4-dev' into 5.0-dev

bembelimen committed:
Bump version to 5.0.0-RC2

bembelimen committed:
Reset to dev

hans2103 commented:
> The width/height settings are ignored.
> 
> Before PR:
> 
> ```
> <img
>                                 src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg#joomlaImage://local-images/banners/shop-ad-books.jpg?width=468&height=60"
>                                 alt="Banner"
>                                 width="50"                                height="50"                            >
> ```
> 
> After PR:
> 
> ```
> <img src="http://localhost/Joomla_5.0.0-alpha2-Alpha-Full_Package/images/banners/shop-ad-books.jpg" alt="Banner" width="468" height="60" loading="lazy">
> ```

are they? The attributes `width` and `height` are still there.

brianteeman committed:
[4.4] Noupdate available to joomla 5 (#42103)

* typo

* [4.4] No Update message

* oops

* oops

* variable

MacJoom committed:
Joomla! 4.4.0 Stable

HLeithner committed:
Merge branch '4.4-dev' into 5.0/upmerge/231014

HLeithner committed:
Merge pull request #42137 from HLeithner/5.0/upmerge/231014

[5.0] Upmerge 4.4

HLeithner committed:
Joomla! 5.0.0 Stable

Signed-off-by: Harald Leithner <leithner@itronic.at>

HLeithner committed:
reset to dev

MacJoom committed:
Revert to dev

Signed-off-by: Martin Kopp <kopp05@infotech.ch>

HLeithner committed:
[5.0] Update build script for new github layout (#42152)

* Add zstandard and github layout

Make zstandard a default compression method and demote bzip2
Also make github layout more user friendly

* cs

brianteeman committed:
[5.0] installerhelper codestyle (#42138)

brianteeman committed:
[5.0] com_joomlaupdate comment (#42139)

Simple pr to update the file referenced in a todo

codereview

dgrammatiko committed:
[5.0] Media manager list view parity (#42119)

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

Hackwar committed:
Custom Fields: Add warning to Smart Search option about information disclosure (#42111)

Hackwar committed:
[5.0] MailTemplate: Prevent notice in replaceTags (#41679)

* MailTemplate: Prevent notice in replaceTags

* Update libraries/src/Mail/MailTemplate.php

Co-authored-by: Quy <quy@nomonkeybiz.com>

* Update MailTemplate.php

---------

Co-authored-by: Quy <quy@nomonkeybiz.com>

wilsonge committed:
Rework the com_messages message view (#42135)

Co-authored-by: Quy <quy@nomonkeybiz.com>

wilsonge committed:
Remove template special color from core use (#42167)

dgrammatiko committed:
[5.0] Fixes the cli installer (#42116)

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Harald Leithner <leithner@itronic.at>

bembelimen committed:
Update min PHP requirement to 8.1 (#42174)

HLeithner committed:
Add proper error handling for public folder cli installation (#42168)

Co-authored-by: Quy <quy@nomonkeybiz.com>

OctavianC committed:
Import behaviour plugins (#42217)

dgrammatiko committed:
[5.0][bug] TinyMCE Setting linebreak with doesn´t work (#42227)

* Update DisplayTrait.php

* Update plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php

MacJoom committed:
Context Issue 41136 (#41150)

brianteeman committed:
[4.4] Remove duplicate png (#41900)

* [4.3] Remove duplicate png

simple pr to correct an error from #34634 where png was added twice

* Update base.sql

* Update base.sql

Hackwar committed:
[4.4] Fixing drone packager upload (#42247)

* Fixing drone package upload

* Adding notification for nightly builds to mattermost

* Further fixes

* Fixing name

alikon committed:
related_items (#42245)

alikon committed:
[4] article field List on webservices (#42206)

* article fieldList on webservices

* Update plugins/fields/list/src/Extension/ListPlugin.php

i don't like elseif

Co-authored-by: Quy <quy@nomonkeybiz.com>

---------

Co-authored-by: Quy <quy@nomonkeybiz.com>

chmst committed:
[4.4.0] Fix condition for menuitems (#42157)

Co-authored-by: Quy <quy@nomonkeybiz.com>

kochinc committed:
[4.4] Proposed fix for Workflow transition error (#42036)

* Update UsersModel.php for 'GROUP BY' fix

* Update administrator/components/com_users/src/Model/UsersModel.php

Conform to coding style

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update Workflow.php for issue #42031

Proposed fix for the reported error at #42031:
An error has occurred.
42601 42601, 7, ERROR: syntax error at or near "WHERE" LINE 5: WHERE "t"."id" = $1 AND "t"."workflow_id" = "w"."id" AND "t"... ^

* Revised update Workflow.php for #42031

Corrected join() call.

* Update Workflow.php for code style

Remove a space which caused phpcs complained.

---------

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Quy <quy@nomonkeybiz.com>

MacJoom committed:
Fix for issue 42229 - webservices content patch tags lost (#42233)

* Fix for issue 42229 - webservices content patch tags lost

* cs

SniperSister committed:
[5.0] Update TinyMCE to 6.7.2 (#42240)

* Update tinymce to 6.7.2

* update plugin manifest version

Fedik committed:
User field: fix "No user" selection (#42256)

Fedik committed:
Fix j+a shortcut in TinyMCE JoomlaHighlighter plugin (#42258)

dgrammatiko committed:
Update plugin.es6.js (#42263)

LadySolveig committed:
[5.0] Fix and Update MySQL to Version 8.0 AppVeyor (#42267)

* [5.0] Fix and Update MySQL to Version 8.0 AppVeyor

* Fix formatting

SniperSister committed:
[4.4] Update TinyMCE to 5.10.8 (#42239)

* Update TinyMCE to 5.10.8

* update plugin manifest version

Hackwar committed:
[4.4] Appveyor: Raise MySQL to version 8.0 (#42271)

Hackwar committed:
Fixing packaging and nightly builds (#42273)

Hackwar committed:
Drone: Renaming build script (#42274)

AndySDH committed:
Fix for Finder (Smart Search) indexer removing spaces in long strings (#41937)

svanschu committed:
fix #42191 Unknown column 'id' in 'field list' (#42199)

In this method it tries to load the field 'id' from the extensions table. But there isn't any field called id in this table. The id field for the extenions table is called extension_id

Denitz committed:
Update Router.php (#41296)

Thank you

wilsonge committed:
Fix new J4 components missing assets (#41358)

OctavianC committed:
Fix multi-factor migration when updating from J3 to J4 (#42253)

* Update MultiFactorAuthenticationHandler.php

* Remove unuised MfaTable

---------

dgrammatiko committed:
[5.0] Cassiopeia should respect inheritance (#42294)


---------

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

Fedik committed:
[5.0] Filter values of custom field, fix #42259 (#42285)


---------

Co-authored-by: Quy <quy@nomonkeybiz.com>

Fedik committed:
Fix tinymce error on mobile (#42306)

Fedik committed:
TinyMCE add missing event listener (#42321)

janschoenherr committed:
Fix missing archive icon (#42316)

* Fix missing icon archive

This PR fixes a php warning for archived articles.

* Update administrator/components/com_content/tmpl/articles/modal.php

---------

Co-authored-by: Quy <quy@nomonkeybiz.com>

alikon committed:
Saving User Profile logs you out of the backend (#42284)

Fedik committed:
Fix field Color (Simple) rendering (#42323)

C-Lodder committed:
Improve build times (#42325)

dgrammatiko committed:
[4.4] Node 20 (#42331)

* Node 20

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

* Update build.js

* sign drone config

---------

Signed-off-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Fedik committed:
Fix table multiselect (#42330)

* Multiselect update

* Multiselect update

* Multiselect update

* Multiselect update

* Multiselect update

* Update build/media_source/system/js/multiselect.es6.js

Co-authored-by: Dimitris Grammatikogiannis <dg@dgrammatiko.dev>

Fedik committed:
Fix list select (with choicesjs) losing values in subform (#42332)

C-Lodder committed:
Don't force the process to terminate (#42341)

Denitz committed:
str_replace() on NULL (#40528)

bembelimen committed:
Merge remote-tracking branch 'Joomla/4.4-dev' into 5.0/upmerge-2023-11-12

bembelimen committed:
Update drone yml

heelc29 committed:
[4.4] translation description text static error pages (#42030)

* move meta declaration

for better upmerge (conflicts #41935)

* same help link element

* check if valid ref

* check if valid elements

* add descText1

* update static error pages

* use early return

Fedik committed:
Text::script should update existing script strings (#41543)

* Text::script should update existing script strings

* Text::script should update existing script strings

* comment

bembelimen committed:
Merge pull request #42345 from bembelimen/5.0/upmerge-2023-11-12

[5.0] Upmerge 2023-11-12

carlitorweb committed:
Admin Articles View - Execute transition button permissions fixed (#39490)

* Admin Articles View - Transition button fixed

* fixed drone claim

---------

brianteeman committed:
[5.0]Spelling templates (#42350)

simple spelling correction

brianteeman committed:
[5.0] spelling descending (#42348)

Simple spelling correction

Co-authored-by: Benjamin Trenkle <bembelimen@users.noreply.github.com>
Co-authored-by: Quy <quy@nomonkeybiz.com>

brianteeman committed:
[5.0] Spelling mismatch (#42349)

Simple spelling correction

Co-authored-by: Benjamin Trenkle <bembelimen@users.noreply.github.com>
Co-authored-by: Quy <quy@nomonkeybiz.com>

dgrammatiko committed:
Use .env files (#42340)

Fedik committed:
Fix a bug of web assets file joomla.asset.json registration/parsing (#41431)

* Make sure that all registry files parsed

* Make sure that all registry files parsed

---------

bembelimen committed:
Merge remote-tracking branch 'Joomla/4.4-dev' into 5.0/upmerge-2023-11-17

HLeithner committed:
[5.0] Fix legacy table creation (#42180)

* Fix legacy table creation

If a legacy table is created by Table::getInstance() twice it fails the second time. Reason for this is that the legacy table name is not used if the table class already exists.

bembelimen committed:
Revert lock file version

bembelimen committed:
Merge branch '5.0-dev' into 5.0/upmerge-2023-11-17

bembelimen committed:
Merge pull request #42360 from bembelimen/5.0/upmerge-2023-11-17

[5.0] upmerge 2023-11-18

wilsonge committed:
Add basic support for conditional dark mode themes in our editors (#42322)

Fedik committed:
Fix URLs and configurable keyName for editing (#42346)

laoneo committed:
Declare the context variable correctly (#42336)

* Declare the context variable correctly

* cs

---------

MacJoom committed:
Joomla! 4.4.1 Rc 1

bembelimen committed:
Update TinyMCE

bembelimen committed:
Bump version

MacJoom committed:
Revert to dev

Signed-off-by: Martin Kopp <kopp05@infotech.ch>

bembelimen committed:
Reset to dev

bembelimen committed:
Fix sass-embedded dependencies (#42369)

brianteeman committed:
[4.4] TinyMCE 5.10.9 (#42359)

This is a security release

## Version 5.10.9 - November 15, 2023
### Changed
- Zero width no-break space (U+FEFF) characters are removed from content passed to setContent, insertContent, and resetContent APIs.
- Zero width no-break space (U+FEFF) characters in initial content are not loaded into the editor upon initialization.
### Fixed
-Specific HTML content containing unescaped text nodes caused mXSS when using undo/redo.
-Specific HTML content containing unescaped text nodes caused mXSS when using the getContent and setContent APIs with the format: 'raw' option, which also affected the resetContent API and the draft restoration feature of the Autosave plugin

MacJoom committed:
Joomla! 4.4.1 Rc 2

MacJoom committed:
Joomla! 4.4.1 Stable

Fedik committed:
Revert TinyMCE builder dark skin (#42392)

brianteeman committed:
[5.0] revert string removal (#42388)

brianteeman committed:
[5.0] Correct language string workflows (#42389)

bembelimen committed:
Joomla 5.0.1 Release Candidate 2

bembelimen committed:
Reset to dev

bembelimen committed:
Joomla! 5.0.1 Stable

MacJoom committed:
Revert to dev

Signed-off-by: Martin Kopp <kopp05@infotech.ch>

bembelimen committed:
Reset to dev

brianteeman committed:
[4.4] Broken language comment (#42425)

* Broken language comment

The recent security update has exposed a bug in a language string. You cannot have a " in a comment.

* dont translate images

Denitz committed:
[4.4] json_decode of NULL in InstallerScript::getItemArray() (#40527)

* json_decode of NULL

* Check for name in Installer Script

Hackwar committed:
Finder: Do not show error messages for suggestions (#42490)

Hackwar committed:
com_contact: Fixing ignored loadData parameter for getForm (#42491)

chmst committed:
[4.4] Fix error editor not found in contenthistory modal (#42458)

* Fix error editor not found in contenthistory modal

* cs

* update as required

bembelimen committed:
Merge remote-tracking branch 'Joomla/4.4-dev' into 5.0/upmerge-2023-12-12

bembelimen committed:
Merge pull request #42497 from bembelimen/5.0/upmerge-2023-12-12

[5.0] Upmerge 2023-12-12

alikon committed:
deprecation php 8.2 (#42451)

hans2103 committed:
Merge branch 'upstream/5.0-dev' into feature/5.0-dev/replace-img-by-jlayout-for-mod-banners

hans2103 commented:
please close this PR as it is replaced by https://github.com/joomla/joomla-cms/pull/42508
